stdlib: create an init function for records with complex default values (#9373)
erlang/otp: system/doc/reference_manual/ref_man_records.md

> using the syntax described in this section. In earlier releases, it was
> necessary to use the `m:shell` built-in function `rd/2`.

> #### Change {: .info }
>
> In Erlang/OTP 28, variables are allowed in the record definition.

> > If a record definition contains a variable, a function call will be made to
> > initialize the records omitted fields.

Reviewer: Isn't the function call an implementation detail? Why should it be mentioned? I think that it would be to describe the semantics.

```suggestion
> If a record definition contains a variable, the compiler will insert a function call to
> initialize the record's omitted fields.
```

```erlang
-record(r, {a = case my_f() of {X, _} -> X; _ -> ok end,
            b = case my_g() of {Y, _} -> Y; _ -> ok end}).
t(X) ->
   #r{}. %% X will not have any effect on initialization of #r
```

> Variables bound in a field are visible in the fields defined after
> so use with caution.

Reviewer: I thought that each field would be independent. This seems error-prone.

```suggestion
Variables bound in a field are visible in subsequent fields, so use them with caution.
```

Author: Same behavior as if you would define a tuple, chosen to have it this way because of consistency
{r, case my_f() of {X, _} -> X; _ -> ok end, case my_g() of {X, _} -> X; _ -> ok end}

```erlang
-record(r, {a = case my_f() of {Res, _} -> Res; _ -> ok end,
            b = case my_g() of {Res, _} -> Res; _ -> error end}).
```

> Variables need to be bounded, which is not the case in the following.

Reviewer: Are you trying to say that the following example will cause a compiler error? If so, I suggest that you show the compiler error.

```suggestion
Variables need to be bound, which is not the case in the following.
```

definition, and fields can be omitted. Omitted fields get their respective
default value instead.

> #### Change {: .info }
>

> > In Erlang/OTP OTP-19464, variables are allowed in the record definition.
> > If a record definition contains a variable, a function call will be made to
> > initialize the records omitted fields.

Reviewer: Note that ticket references must written like `@OTP-19461@` in order to get them replaced with the version. In this case, I suggest you just write `28`.

```suggestion
> From Erlang/OTP 28 variables are allowed in record definitions.
> Variables defined and used in an expression for initializing a record field are local to that expression.
```

I suggest that you add a few examples to show that will work and not work.
